### PR TITLE
Protect db threads

### LINKED ISSUE #400: Protect threads that are backing a db from direct deletion
Otherwise, on re-load, this happens:
```
textile_1  | 2020-07-14T01:23:39.912Z	FATAL	db	db/db.go:173	unable to connect app: error getting thread bafk5omesrcb4nmnclmrqzhufzmz6manujcpuvpnei7jlstfnhcd7djq: thread not found
```

### PR DESCRIPTION
**Pull request by sanderpick**
Closes #400

- Adds a thread maps `apps` to `net`, which is used to determine if a thread is connected to an app and protect it from deletion 
- Skips the fatal log when there's an error connecting an app to a thread and deletes the failing db (needed because production has some orphaned dbs)
- Forces a db to have a name, since without a name, there won't be any keys in the datastore to reload the db with after a manager restart... but the underlying thread will still be there :/

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -57,33 +57,3 @@ jobs:
           go get -v -t -d ./...
       - name: Test
         run: go test ./integrationtests/foldersync/
-
-  android-client:
```
Comment (sanderpick) [inline .github/workflows/test.yml:61]: Dropping these tests... we don't have official support anymore

File: db/manager_test.go [hunk 2]
```diff
@@ -172,12 +172,35 @@ func TestManager_GetDB(t *testing.T) {
...
+		checkErr(t, err)
+
 		time.Sleep(time.Second)
 
 		err = man.Close()
 		checkErr(t, err)
 		err = n.Close()
 		checkErr(t, err)
+
+		t.Run("test get deleted db after restart", func(t *testing.T) {
```
Comment (sanderpick) [inline db/manager_test.go:186]: This is just double double checking that a db is actually deleted after deleting it.

File: db/manager.go [hunk 3]
```diff
@@ -83,16 +84,28 @@ func NewManager(network app.Net, opts ...NewOption) (*Manager, error) {
...
+			continue
+		}
 		opts, err := getDBOptions(id, m.opts, "")
 		if err != nil {
 			return nil, err
 		}
 		s, err := newDB(m.network, id, opts)
 		if err != nil {
-			return nil, err
+			log.Errorf("unable to reload db %s: %s (marked for deletion)", id, err)
```
Comment (carsonfarmer) [inline db/manager.go:96]: 👍

File: core/app/app.go [hunk 4]
```diff
@@ -169,6 +172,11 @@ func (c *Connector) ThreadID() thread.ID {
 	return c.threadID
 }
 
+// Token returns the net token.
+func (c *Connector) Token() net.Token {
```
Comment (sanderpick) [inline core/app/app.go:176]: Adding the concept of a token to a connector

File: core/app/app.go [hunk 5]
```diff
@@ -212,7 +220,7 @@ func (c *Connector) appToThread(wg *sync.WaitGroup) {
 				return
 			}
 			ctx, cancel := context.WithTimeout(context.Background(), addRecordTimeout)
-			if _, err := c.Net.CreateRecord(ctx, c.threadID, event.Node, net.WithThreadToken(event.Token)); err != nil {
+			if _, err := c.Net.CreateRecord(ctx, c.threadID, event.Node, net.WithThreadToken(event.Token), net.WithAPIToken(c.token)); err != nil {
```
Comment (sanderpick) [inline core/app/app.go:223]: Apps (dbs) can supply their token to the net api to authenticate

File: core/net/net.go [hunk 6]
```diff
@@ -69,3 +70,12 @@ type API interface {
 	// Subscribe returns a read-only channel of records.
 	Subscribe(ctx context.Context, opts ...SubOption) (<-chan ThreadRecord, error)
 }
+
+// Token is used to restrict network APIs to a single app.App.
+// In other words, a net token protects against writes and deletions
+// external to an app.
+type Token []byte
```
Comment (sanderpick) [inline core/net/net.go:77]: See docstring

**Pull request merged**